### ISSUE TITLE
Build/test jenkins infrastructure review

### ISSUE THREAD

**Issue by jdekonin** [eclipse/omr#6501]
**Current state of available test nodes.**
2 * AIX v7.1
2 * Windows 2012 (1 is dead)
3 * Mac x86
13 * pLinux; 10 * ub16 @ UNB, 3 * ub16 @ OSU (2 for build/test and 1 is a proxy to access UNB systems)
13 * xLinux; 12 * ub16 @ UNB, 1 * debian 10 @ UNB (riscv x-compile)
2 * zLinux; both ub16 @ Marist
2 * z/OS v2.4

**Suggestions, in no specific order of importance**
1) Replace existing Windows 2012 with 2 Windows 2019...upgrade the version of Visual Studio to match lowest support requirement for OpenJ9 which is for JDK8, currently VS2013.
2) Upgrade 3 Macs to lowest support requirement for OpenJ9 which is 10.15.7 and xcode 10.3
3) Convince Pete/OpenJ9 community to swap a Mac x86 for a Mac M1.  Currently no Mac M1 coverage is present.
4) Add 2 * aLinux (aarch64) ub20 systems.  Currently no aLinux coverage is present.
5) Upgrade zLinux systems to ub20
6) Upgrade pLinux systems
  a) 3 @ OSU to ub20
  b) 10 @ UNB to ub20
  c) when ub22 becomes an option upgrade 5 of the ub20 systems
7) Upgrade xLinux systems
  a) 6 to ub20 with cgroup v1 support
  b) 6 to ub20 with cgroup v2 support
  c) when ub22 becomes an option upgrade 6 of the ub20 systems

Note this does not include any coverage via Azure which currently exists for Windows, xLinux and Mac x86.  I am not sure what the limitations are with Azure resources, but if OMR isn't hitting any limits freeing up Windows systems to be used at OpenJ9 could be an option.

Related cgroup support raised via #6468

@0xdaryl @babsingh please add others that would have an stake in these systems being updated.  I'm looking for agreement to an more up-to-date set of machines on the OMR jenkins build system.

**Comment by babsingh** [comment]
GC: @amicic @dmitripivkine 
VM: @tajila @pshipton 
JIT: @mstoodle @dsouzai

**Comment by dsouzai** [comment]
I think overall the plan makes sense. Couple of questions:

1. Do we anticipate a need to keep a couple of older OSes for some kind of compatibility testing?
2. Is there a plan to add cgroup v2 support for the non x86 linux machines?

**Comment by babsingh** [comment]
> Is there a plan to add cgroup v2 support for the non x86 linux machines?

cgroup v2 support is not architecture dependent. cgroup v2 support on non x86 Linux machines won't add any new value or coverage. x86 linux machines should be sufficient for cgroup v2 support.

**Comment by jdekonin** [comment]
I've done 2 for starters to get @babsingh unblocked with his cgroup work.  @babsingh and I have both tested these systems and they both work for existing workloads so I am not worried there will be unexpected failures with OMR builds.  I'll advertise in slack via #build-infra as well for awareness of the change.

**Comment by 0xdaryl** [comment]
@jdekonin : this re-provisioning looks reasonable to me.

I think @AdamBrousseau mentioned at one point that acquiring more alinux nodes from OSU might be a possibility as well.

**Comment by jdekonin** [comment]
> ...acquiring more alinux nodes from OSU might be a possibility as well.

Yes that was one of the reasons I mentioned it, but capacity from Equinix will also be increased in the near future :)

The updates will slowly be worked through and be announced in the build-infra slack channel as the occur.

**Comment by babsingh** [comment]
> Do we anticipate a need to keep a couple of older OSes for some kind of compatibility testing?

Just looking at Ubuntu: https://www.thomas-krenn.com/en/wiki/Ubuntu_LTS_Hardware_Enablement_Stack_information.

Only Ub18 and Ub20 are currently supported. Ub16 is no longer supported. _Newer Ubuntu releases try to be as much backward-compatible as possible._ If OMR compilation and testing works on the latest Ub20, it should also work on Ub18 and Ub16. At least for Ubuntu, I don't believe older OSes are needed for compatibility testing. 

Not sure about other OSes. A similar evaluation can be done for other OSes.

**Comment by jdekonin** [comment]
hosted on kvm1:
- ub20-x64-omr1, ub20-x64-omr3, ub20-x64-omr5 are cgroup.v1
- ub20-x64-omr7, ub20-x64-omr9, ub20-x64-omr11 are cgroup.v2

hosted on kvm2:
- ub20-x64-omr8, ub20-x64-omr10, ub20-x64-omr12 are cgroup.v1
- ub20-x64-omr2, ub20-x64-omr4, ub20-x64-omr6 are cgroup.v2

PR for updating the jenkins nodes
https://github.com/eclipse-cbi/jiro/pull/215

**Comment by AdamBrousseau** [comment]
Had to update etc/hosts on the jumpbox and add known host entries for the new hostnames. Machines are online now (ub20-x64-[3-12])
